gateway/src/main/java/com/hurence/webapiservice/http/grafana/AnnotationRequestParser.java

```diff
@@ -0,0 +1,74 @@
+package com.hurence.webapiservice.http.grafana;
+
+import com.hurence.webapiservice.http.grafana.modele.AnnotationRequestParam;
+
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+import static com.hurence.webapiservice.http.grafana.util.DateRequestParserUtil.*;
+
+public class AnnotationRequestParser {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(QueryRequestParser.class);
+
+
+
+    public AnnotationRequestParam parseAnnotationRequest(JsonObject requestBody) throws IllegalArgumentException {
+        LOGGER.debug("trying to parse annotation requestBody : {}", requestBody);
+        AnnotationRequestParam.Builder builder = new AnnotationRequestParam.Builder();
+        long from = parseFrom(requestBody);
+        builder.from(from);
+        long to = parseTo(requestBody);
+        builder.to(to);
+        long fromRaw = parseFromRaw(requestBody);
+        builder.fromRaw(fromRaw);
+        long toRaw = parseToRaw(requestBody);
+        builder.toRaw(toRaw);
+        int maxAnnotations = parseMaxAnnotations(requestBody);;
+        builder.withMaxAnnotation(maxAnnotations);
+        JsonArray tags = parseTags(requestBody);;
+        builder.withTags(tags);
+        Boolean matchAny = parseMatchAny(requestBody);;
+        builder.withMatchAny(matchAny);
+        String type = parseType(requestBody);
+        builder.withType(type);
+        return builder.build();
+
+    }
+
+    private long parseFromRaw(JsonObject requestBody) {
+        return parseDate(requestBody, "/rangeRaw/from");
```

Comment:
Did you look to /rangeRaw/from value ? 
```json
  "rangeRaw": {
    "from": "now-1h",
    "to": "now"
  }
```
This parsing would fail... You do not need to parse that relative time. You should add unit test that parse a request to ensure your parser works fine.